conans/client/generators/__init__.py

```diff
@@ -28,6 +28,7 @@
 from conans.util.env_reader import get_env
 from .b2 import B2Generator
 from .premake import PremakeGenerator
+from .make import MakeGenerator
```

Comment:
to be honest, I don't think generic name `MakeGenerator` is the best choice
I suppose it's better to make it e.g. `GNUMakeGenerator` to indicate it works for GNU Make.
(you know, there are lots of incompatible make derivatives, like BSD Make, NMake, etc. and so far it's not tested with them all, but only GNU Make, right?)

Reply:
On second thought, this raises an interesting point.  Because this generator only assigns values to variables using the most basic of operators, I believe it might be universally compatible.  Internally, we have a `gnu_make_generator` and `opus_make_generator` which are VERY different, and produce rules and conditionals which are incompatible with each other, but both support the basic assignment operators. 

As such, I now actually think `MakeGenerator` might be the most appropriate name.

Reply:
- The internal name of the generator class could be changed in the future, what it is important here is the name of the user interface. Of this one it would be ``generators = "make"``. Should we provide a different one no avoid confusion? ``gnumake``? ``generic_make``?

- @solvingj Are you saying that this generator not compatible with GNU Make?

Reply:
sorry to be unclear.  

This does work with GNU Make.  I was saying that in theory, it should also work for any Make variant. 

RE: naming, I was referring to the generators name for usage.

`make_generic` is probably what i would recommend right now, but i have to say i don't like it, or any of the other options.  I could also accept simply `make` since it really is the most generic and universal `make` generator there will ever be.  Also, it's actually NOT specific to `gnu_make`, as it's known to work with multiple makes (tested with Opus Make).  So, `gnu_make` seems wrong. 
 It would be nice to prove it works with `NMake` in some future test, but that's not required.

So, in summary:  `make` or `generic_make` or `make_generic` all seem fine.  Now i'm leaning towards simply `make` again.

I'll leave it up to Conan team to decide :)

Reply:
We will leave it as is. Thanks a lot for the feedback and clarification 😄